### PR TITLE
Update doxygen comments in OTA agent header

### PR DESCRIPTION
**Pull request by tgsong**
Description
-----------

Update the doxygen comments to make OTA conform to C-SDK doxygen style. Some parts are rearranged to make it better organized, no real code change.

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [ ] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: libraries/freertos_plus/aws/ota/include/aws_iot_ota_agent.h [hunk 1]
```diff
@@ -451,7 +371,75 @@ typedef int16_t (* pxOTAPALWriteBlockCallback_t)( OTA_FileContext_t * const C,
...
+ */
+
+/* A composite cryptographic signature structure able to hold our largest supported signature. */
+
+#define kOTA_MaxSignatureSize    256        /* Max bytes supported for a file signature (2048 bit RSA is 256 bytes). */
+
+typedef struct
+{
+    uint16_t usSize;                         /* Size, in bytes, of the signature. */
+    uint8_t ucData[ kOTA_MaxSignatureSize ]; /* The binary signature data. */
```
Comment (dcgaws) [inline libraries/freertos_plus/aws/ota/include/aws_iot_ota_agent.h]: This means that we're always using 256 bytes of storage, even though our majority use case is ECDSA. I've typically seen this situation better handled by defining a zero-length array as the last element in the typedef.

File: libraries/freertos_plus/aws/ota/include/aws_iot_ota_agent.h [hunk 2]
```diff
@@ -451,7 +371,75 @@ typedef int16_t (* pxOTAPALWriteBlockCallback_t)( OTA_FileContext_t * const C,
...
+ * @ingroup ota_datatypes_structs
+ * @brief OTA File Context Information.
+ *
+ * Information about an OTA Update file that is to be streamed. This structure is filled in from a
+ * job notification MQTT message. Currently only one file context can be streamed at time.
+ */
+typedef struct OTA_FileContext
+{
+    uint8_t * pucFilePath; /*!< Local file pathname. */
+    union
```
Comment (dcgaws) [inline libraries/freertos_plus/aws/ota/include/aws_iot_ota_agent.h]: This union seems like it should be collapsed into a single PVOID.

File: libraries/freertos_plus/aws/ota/include/aws_iot_ota_agent.h [hunk 3]
```diff
@@ -451,7 +371,75 @@ typedef int16_t (* pxOTAPALWriteBlockCallback_t)( OTA_FileContext_t * const C,
...
+        #endif
+        uint8_t * pucFile;      /*!< File type is RAM/Flash image pointer after file is open for write. */
+    };
+    uint32_t ulFileSize;        /*!< The size of the file in bytes. */
+    uint32_t ulBlocksRemaining; /*!< How many blocks remain to be received (a code optimization). */
+    uint32_t ulFileAttributes;  /*!< Flags specific to the file being received (e.g. secure, bundle, archive). */
+    uint32_t ulServerFileID;    /*!< The file is referenced by this numeric ID in the OTA job. */
+    uint8_t * pucJobName;       /*!< The job name associated with this file from the job service. */
+    uint8_t * pucStreamName;    /*!< The stream associated with this file from the OTA service. */
+    Sig256_t * pxSignature;     /*!< Pointer to the file's signature structure. */
```
Comment (dcgaws) [inline libraries/freertos_plus/aws/ota/include/aws_iot_ota_agent.h]: Move this member to the end of the structure?

File: libraries/freertos_plus/aws/ota/include/aws_iot_ota_agent.h [hunk 4]
```diff
@@ -451,7 +371,75 @@ typedef int16_t (* pxOTAPALWriteBlockCallback_t)( OTA_FileContext_t * const C,
...
+    uint32_t ulUpdaterVersion;  /*!< Used by OTA self-test detection, the version of FW that did the update. */
+    bool_t xIsInSelfTest;       /*!< True if the job is in self test mode. */
+    uint8_t * pucProtocols;     /*!< Authorization scheme. */
+} OTA_FileContext_t;
+
+/**
+ * @ingroup ota_datatypes_structs
+ * @brief OTA Connection context.
+ *
+ * Connection information that user provides to initialize control and data transfer for OTA.
```
Comment (dcgaws) [inline libraries/freertos_plus/aws/ota/include/aws_iot_ota_agent.h]: "the user"

**Comment by tgsong** [comment]
@dcgaws This PR is only for updating the doxygen comments, some part might appear as new in the diff view but actually they're just rearranged with no code change. I can discuss your feedback with @pvyawaha and do another PR to address them.

**Comment by dcgaws** [comment]
I'm fine with tracking the other feedback for a separate PR. Can you please open an Issue so that it doesn't get lost?

**Comment by tgsong** [comment]
Created an sim issue, https://sim.amazon.com/issues/TS-11646. @pvyawaha I assigned this to you, please review.

**Pull request merged**